### PR TITLE
feat: Add API version for v2 DTO

### PR DESCRIPTION
**Pull request by weichou1229**
Since APP service may receive v1 or v2 event DTO from CoreData, we should specify the API version in the DTO.
Fix: #254

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://wiki.edgexfoundry.org/display/FA/Contributor%27s+Guide
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #254 

## What is the new behavior?
* Add API version for v2 DTO
* Add func to convert event and reading from model to DTO

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

### CODE REVIEW

File: v2/dtos/reading.go [hunk 1]
```diff
@@ -65,3 +67,37 @@ func ToReadingModel(r BaseReading, device string) models.Reading {
 	}
 	return readingModel
 }
+
+func FromReadingModelToDTO(reading models.Reading) BaseReading {
+	var baseReading BaseReading
+	switch reading.(type) {
```
Comment (cloudxxx8) [inline v2/dtos/reading.go:73]: use 
`switch r := reading.(type)`
Then, you don't need to cast in each case
https://tour.golang.org/methods/16

Comment (weichou1229) [reply v2/dtos/reading.go]: Modified.

File: v2/dtos/reading.go [hunk 2]
```diff
@@ -5,7 +5,9 @@
 
 package dtos
 
-import "github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+import (
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+)
```
Comment (lenny-intel) [inline v2/dtos/reading.go:10]: In Core WG we decided that all DTOs need to be versioned. So please add version to Reading. Make sense that Request DTOs don't need to be versioned.

Comment (weichou1229) [reply v2/dtos/reading.go:11]: Version added.

**Pull request merged**